tests/Integration/MartinGeorgiev/Doctrine/DBAL/Types/LtreeTypeTest.php

```diff
@@ -0,0 +1,99 @@
+
+final class LtreeTypeTest extends TestCase
+{
+    #[\Override]
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->createLtreeExtension();
+    }
+
+    protected function getTypeName(): string
+    {
+        return 'ltree';
+    }
+
+    protected function getPostgresTypeName(): string
+    {
+        return 'LTREE';
+    }
+
+    #[Test]
+    public function can_handle_null_values(): void
+    {
+        $typeName = $this->getTypeName();
+        $columnType = $this->getPostgresTypeName();
+
+        $this->runDbalBindingRoundTrip($typeName, $columnType, null);
+    }
+
+    #[Test]
+    public function can_handle_string_values(): void
+    {
+        $typeName = $this->getTypeName();
+        $columnType = $this->getPostgresTypeName();
+
+        $this->runDbalBindingRoundTrip($typeName, $columnType, 'root.child.grand-child');
+    }
+
+    /**
+     * Override to handle Ltree value object comparison.
+     */
+    protected function assertTypeValueEquals(mixed $expected, mixed $actual, string $typeName): void
+    {
+        if (!$actual instanceof LtreeValueObject) {
+            throw new \InvalidArgumentException('LtreeTypeTest expects actual value to be a Ltree object');
+        }
+
+        if (!$expected instanceof LtreeValueObject && !\is_string($expected)) {
+            throw new \InvalidArgumentException('LtreeTypeTest expects expected value to be a Ltree object or string');
+        }
+
+        $this->assertLtreeEquals($expected, $actual, $typeName);
+    }
+
+    #[DataProvider('provideValidTransformations')]
+    #[Test]
+    public function can_handle_ltree_values(LtreeValueObject $ltreeValueObject): void
+    {
+        $typeName = $this->getTypeName();
+        $columnType = $this->getPostgresTypeName();
+
+        $this->runDbalBindingRoundTrip($typeName, $columnType, $ltreeValueObject);
+    }
+
+    /**
+     * @return array<string, array{LtreeValueObject}>
+     */
+    public static function provideValidTransformations(): array
+    {
+        return [
+            'ltree simple string' => [new LtreeValueObject(['foo', 'bar', 'baz'])],
+            'ltree simple numeric' => [new LtreeValueObject(['1', '2', '3'])],
+            'ltree single numeric' => [new LtreeValueObject(['1'])],
+            'ltree empty' => [new LtreeValueObject([])],
+        ];
+    }
+
+    private function assertLtreeEquals(LtreeValueObject|string $ltreeValueObject, mixed $actual, string $typeName): void
+    {
+        $this->assertInstanceOf(LtreeValueObject::class, $actual, 'Failed asserting that value is a Ltree object for type '.$typeName);
+        $this->assertSame((string) $ltreeValueObject, (string) $actual, 'Failed asserting that ltree string representations are identical for type '.$typeName);
+    }
+
+    private function createLtreeExtension(): void
+    {
+        $sql = 'CREATE EXTENSION IF NOT EXISTS ltree';
+        $this->connection->executeStatement($sql);
+    }
```

Comment:
We already centralize extension setup in `tests/Integration/MartinGeorgiev/TestCase.php`. Please hook this into the [createTestSchema()](https://github.com/martin-georgiev/postgresql-for-doctrine/blob/d97caae042ecbdd1e958b512e38ca4d56fd3d711/tests/Integration/MartinGeorgiev/TestCase.php#L168) mechanics instead of duplicating it here. This keeps initialization consistent and reduces maintenance.